### PR TITLE
Added contributor model, repo, service, dto, controller

### PR DESCRIPTION
**Pull request by koylubaevnt**
[EPMLABSBRN-XX](https://jira.epam.com/jira/browse/EPMLABSBRN-XX)

**Description**:

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.epam.brn.dto
+
+data class ContributorAdminDto(
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt:3]: not clear what you mean by Admin

if by use case to add/edit - so it should be
dto/request/ContributorRequest

File: src/main/kotlin/com/epam/brn/dto/ContributorUserDto.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.epam.brn.dto
+
+data class ContributorUserDto(
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/ContributorUserDto.kt:3]: we are waiting as I understand
ContributorDevelopersResponse
and
ContributorSpesialistsResponse

File: src/main/kotlin/com/epam/brn/dto/GitHubUserDto.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.epam.brn.dto
+
+data class GitHubUserDto(
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/GitHubUserDto.kt:3]: so let it be like in enum
ContributorDeveloperResponse
ContributorSpecialistResponse

File: src/main/kotlin/com/epam/brn/model/Contributor.kt [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    var name: String? = null,
+    var description: String? = null,
+    var company: String? = null,
+    var nameEn: String? = null,
+    var descriptionEn: String? = null,
+    var companyEn: String? = null,
+    @Enumerated(EnumType.STRING)
+    var type: ContributorType = ContributorType.DEVELOPER,
+    var pictureUrl: String? = null,
+    var contribution: Long? = null,
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/model/Contributor.kt:35]: Long?
->
Long = 0,

File: src/main/kotlin/com/epam/brn/service/ContributorServer.kt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.epam.brn.service
+
+import com.epam.brn.dto.ContributorUserDto
+import com.epam.brn.enums.ContributorType
+
+interface ContributorServer {
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/service/ContributorServer.kt:6]: ContributorServer 
->
ContributorService

**Review by ElenaSpb — CHANGES_REQUESTED** [review]
fix naming please

File: src/main/kotlin/com/epam/brn/dto/ContributorUserDto.kt (hunk 2 above)
Comment (koylubaevnt) [reply src/main/kotlin/com/epam/brn/dto/ContributorUserDto.kt:3]: Should we split Developers and Specialist? I think we can use both as one entity.
Different for ContributorUserDto and ContributorAdminDto: they are response, but one for user page and other for admin page...
I don't know how it's well name

File: src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt (hunk 1 above)
Comment (koylubaevnt) [reply src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt:3]: It's response for admin page. I don't know how it's well name

File: src/main/kotlin/com/epam/brn/dto/GitHubUserDto.kt (hunk 3 above)
Comment (koylubaevnt) [reply src/main/kotlin/com/epam/brn/dto/GitHubUserDto.kt:3]: I don't understand what does you mean?
GitHubUserDto its dto for information about github users. I think it's good for admin page to show information about it

File: src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt (hunk 1 above)
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt:3]: yes, I see, 
ContributorDetailsResponse ?
ContributorDetailsRequest for edit/add end-point later
?

File: src/main/kotlin/com/epam/brn/dto/GitHubUserDto.kt (hunk 3 above)
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/dto/GitHubUserDto.kt:3]: oh, yes, so 
dto/request/GitHubUserRequest
?

File: src/main/kotlin/com/epam/brn/dto/ContributorUserDto.kt (hunk 2 above)
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/dto/ContributorUserDto.kt:3]: ->
ContributorResponse
one model for both end-point as fields are mostly the same
?

File: src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt (hunk 1 above)
Comment (koylubaevnt) [reply src/main/kotlin/com/epam/brn/dto/ContributorAdminDto.kt:3]: Need add "ContributorDetailsRequest for edit/add end-point later" in issue where describe admin operation for contributors

File: src/main/kotlin/com/epam/brn/dto/response/ContributorDetailsResponse.kt [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    var type: String,
+    var name: String?,
+    var description: String?,
+    var company: String?,
+    var nameEn: String?,
+    var descriptionEn: String?,
+    var companyEn: String?,
+    var pictureUrl: String?,
+    var contacts: Set<ContactResponse>,
+    var gitHubUser: GitHubUserResponse?,
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/response/ContributorDetailsResponse.kt:14]: if they are only part of Response and do use separatly - looks like it is really Dto.
?
ContactDto
GitHubUserDto

File: src/main/kotlin/com/epam/brn/dto/response/GitHubUserResponse.kt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.epam.brn.dto.response
+
+data class GitHubUserResponse(
+    var id: Long? = null,
+    var name: String? = null,
+    var email: String? = null,
+    var avatarUrl: String?,
+    var bio: String?,
+    var company: String?,
+)
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/response/GitHubUserResponse.kt:10]: why all fields are nullable?
how id can be nullable if the object is saved in our DB?
maybe name is notnullable too?

if Contributor do not have link to githubuser- so 
contributor.githubuser=null
?

File: src/main/kotlin/com/epam/brn/dto/response/ContributorDetailsResponse.kt (hunk 6 above)
Comment (koylubaevnt) [reply src/main/kotlin/com/epam/brn/dto/response/ContributorDetailsResponse.kt:14]: If they DTO then need move to dto folder?

File: src/main/kotlin/com/epam/brn/dto/response/GitHubUserResponse.kt (hunk 7 above)
Comment (koylubaevnt) [reply src/main/kotlin/com/epam/brn/dto/response/GitHubUserResponse.kt:10]: Yes, If contrinutor don't have link to githubuser it was =null.
Id can't be null, I will fix it. I check field from github api, but think all field cant't be null. I will change it

File: src/main/kotlin/com/epam/brn/controller/ContributorController.kt [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import io.swagger.annotations.ApiOperation
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/contributor")
```
Comment (RudzMaksim) [inline src/main/kotlin/com/epam/brn/controller/ContributorController.kt:16]: As I can see by the code base, the plural form is used for REST URLs, so probably it should be '/contributors'

**Review by RudzMaksim — APPROVED** [review]
Just a minor comment about REST URL.
I saw several other places in which I'm not sure but I need to learn the project and codebase before writing comments about them:)

File: src/main/kotlin/com/epam/brn/dto/response/ContributorDetailsResponse.kt (hunk 6 above)
Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/dto/response/ContributorDetailsResponse.kt:14]: yes

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**